crates/rolldown/src/bundler.rs

```diff
@@ -188,12 +189,16 @@ impl Bundler {
 
     self.plugin_driver.generate_bundle(&mut output.assets, is_write).await?;
 
-    output.watch_files = link_stage_output
-      .module_table
-      .modules
-      .iter()
-      .filter_map(|m| m.as_normal().map(|m| m.id.as_str().into()))
-      .collect();
+    output.watch_files = {
+      let mut files = link_stage_output
+        .module_table
+        .modules
+        .iter()
+        .filter_map(|m| m.as_normal().map(|m| m.id.as_str().into()))
+        .collect::<Vec<ArcStr>>();
+      files.extend(self.plugin_driver.watch_files.iter().map(|f| f.clone()));
+      files
+    };
```

Comment:
The rollup also collected the files from `PluginDriver` to `RollupBuild`. ref https://github.com/rollup/rollup/blob/master/src/utils/PluginContext.ts#L56

Reply:
This should be lazy.